dialer.go

```diff
@@ -244,7 +321,9 @@ func (c *Dialer) connectState() {
 		for {
 			select {
 			case c.unreadyCh <- connErr:
-				continue
+				for _, receiver := range c.notifyUnready {
```

Comment:
create a method `func notifyUnready(err error)` that would

* lock
* loop channel,
* do non-blocking channel write

Reply:
the method implemented for both ready that unready